src/utils/with_options/src/lib.rs

```diff
@@ -14,7 +14,34 @@
 
 use proc_macro::TokenStream;
 
-/// Annotates that the struct represents the WITH properties for a connector.
+/// Annotates that the struct represents the `WITH` properties for a connector.
+/// This is a dummy marker that has no effect on the generated code. It's used to generate the `with_option_*.yaml` files.
+///
+/// ## Notes about how to define property structs
+///
+/// ### Prefer strongly-typed fields
+///
+/// Avoid using `HashMap` with `#[serde(flatten)]` to include all unknown fields.
+/// Declare all fields explicitly instead.
+///
+/// The only exception now is CDC, which needs to pass a lot of options as-is to Debezium.
+///
+/// ### `serde` attributes
+///
+/// Always add `#[serde(deny_unknown_fields)]` to the top-level struct.
+/// However, do not add it if there's a `HashMap` field with `#[serde(flatten)]`.
```

Comment:
FIXME

Reply:
We don't rely on `#[serde(deny_unknown_fields)]` any more?

Reply:
Yes, replaced by the hashmap field

Reply:
then how to make the yaml based test work as expected

Reply:
> then how to make the yaml based test work as expected

I don't get it. Isn't it working correctly now?

Reply:
Will you delete everything around `deny_unknown_fields` eventually before merging?

Reply:
Code is already deleted. Only tests and the doc comment here contains it 🤪

Reply:
Emm? How about the `deny_unknown_fields` argument in function signature? such as 

```rust
fn try_from_hashmap(
    props: HashMap<String, String>,
    deny_unknown_fields: bool,
) -> anyhow::Result<Self>
```

I thought this would be removed, too.

Reply:
Maybe I misunderstood this:

> Yes, replaced by the hashmap field

I thought you can check this hashmap to see whether any unknown field appear.
Anyway, either approach LGTM.

Reply:
I meant `#[serde(deny_unkonwn_fields)]` is replaced by the field and the bool flag. 

> I thought you can check this hashmap to see whether any unknown field appear.

Oh, indeed. It seems this doesn't have to be inside `try_from_hashmap`

Reply:
But `try_from_hashmap` is only used by `ConnectorProperties::extract`. I think there's no much difference.